lms/templates/dashboard.html

```diff
@@ -161,17 +161,16 @@
                     # If the user has a fulfilled entitlement, pass through the entitlements CourseEnrollment object
                     enrollment = entitlement_session
                   else:
-                    # If the user has an unfulfilled entitlement, pass through a bare CourseEnrollment object built off of the next available session
-                    upcoming_sessions = course_entitlement_available_sessions[str(entitlement.uuid)]
-                    next_session = upcoming_sessions[0] if upcoming_sessions else None
-                    if not next_session:
+                    # If the user has an unfulfilled entitlement, pass through a bare CourseEnrollment object to populate card with metadata
+                    mock_session = unfulfilled_entitlement_mock_sessions[str(entitlement.uuid)]
+                    if not mock_session:
                         continue
-                    enrollment = CourseEnrollment(user=user, course_id=next_session['key'], mode=next_session['type'])
+                    enrollment = CourseEnrollment(user=user, course_id=mock_session['key'], mode=mock_session['type'])
```

Comment:
Is there anyway that we can move some of this python into the backend?

Reply:
Yeah I tried to move as much as I could into the python, these few lines had to stay unfortunately without reworking the entire page.